### DISCUSSION TITLE
Update Progress Bar toast without animation?

### DISCUSSION THREAD

**Discussion by pcgeek86** [Windos/BurntToast#123] [General] [answered]
When I update a progress bar, it would be nice if the notification toast would just stay in place, instead of animating for each new notification. Is there a way to make that happen? Is there some way to add a `-SkipAnimation` parameter to the `New-BurntToastFunction` command?

```
foreach ($item in 1..5) {
    $Progress = New-BTProgressBar -Status 'Deleting files' -Title 'Cleanup Process' -Value ($item/5) -ValueDisplay 5
    New-BurntToastNotification -ProgressBar $Progress -UniqueIdentifier cleanup -Silent
    Start-Sleep -Milliseconds 5000
}
```

**Comment by Windos** [comment, marked as answer]
This is a cool one, but a bit of a rabbit hole.

What you're seeing atm is the "replace" workflow, where the OS violently rips out the previous notification and displays a brand new one. It can be... jarring if you do them fast enough.

```powershell
foreach ($Number in 1..100) {
    $ProgressBar = New-BTProgressBar -Status 'Running' -Value ($Number / 100)

    $ToastSplat = @{
        Text             = 'Demo Replacement'
        UniqueIdentifier = 'DemoReplace'
        ProgressBar      = $ProgressBar
    }
    New-BurntToastNotification @ToastSplat

    Start-Sleep -Seconds 1
}
```

![Replace](https://user-images.githubusercontent.com/6955786/94737358-34db5a80-03ca-11eb-893a-dd033c12d3df.gif)

Instead of that, what you can do it "update" the existing notification. This requires using "databindings", basically a hashtable of keys (placeholders) and values (the text to replace the placeholders.)

```powershell
# Show your first toast
$FirstDataBinding = @{
    FirstLine = 'Example Toast Heading'
    SecondLine = 'This toast is still the original'
}
New-BurntToastNotification -Text 'FirstLine', 'SecondLine' -UniqueIdentifier 'ExampleToast' -DataBinding $FirstDataBinding

# Wait a bit and update the toast
$SecondDataBinding = @{
    SecondLine = 'This toast has been updated!'
}
Update-BTNotification -UniqueIdentifier 'ExampleToast' -DataBinding $SecondDataBinding
```

![UpdateOnScreen](https://user-images.githubusercontent.com/6955786/94737391-3f95ef80-03ca-11eb-894a-20fe019ba0ca.gif)

You'll note there that I used a unique hashtable the second time around, what I've started doing is reusing the original, so my update might instead be:

```powershell
$FirstDataBinding['SecondLine'] = 'This toast has been updated!'
Update-BTNotification -UniqueIdentifier 'ExampleToast' -DataBinding $SecondDataBinding
```

I've got a blog post on this: https://toastit.dev/2019/07/17/crouton-11/

Also an even more "dynamic" [example](https://github.com/Windos/Talks/blob/main/2020/09%20-%20September/29%20-%20PSPowerHour%20-%20Croque%20Monsieur%20-%20Step%20up%20Your%20Toast%20Notification%20Game/Demos/7_Progress.ps1) from PSPowerHour yesterday, with a [function](https://github.com/Windos/Talks/blob/main/2020/09%20-%20September/29%20-%20PSPowerHour%20-%20Croque%20Monsieur%20-%20Step%20up%20Your%20Toast%20Notification%20Game/Demos/Functions/Start-Progress.ps1) that's doing the updates.

![2020-09-29_14-45-40](https://user-images.githubusercontent.com/6955786/94737604-956a9780-03ca-11eb-856b-f1d7dadc372e.gif)

There are pros and cons of both the replace and the update methods.

Replacing will ALWAYS show the user the new information, even as a brand new toast. e.g. it's half an hour between updates, and you want to alert the user to the updated information.

But the replace isn't great when showing toasts rapidly as you get the flaky animations.

Updating is graceful, and if the toast has gone to the action center it's contents can be updated without pinging onto the users' desktop.

However, you can only update a toast if it exists. If the user dismissed it completely then there is nothing to update.

If you update a toast while it is still on screen, it resets the timeout on it meaning you can keep updating a toast to keep it on screen as long as you're doing it frequently enough.

**Comment by pcgeek86** [comment]
Thanks, yeah I had come across your blog post before. That last example you sent is really what I'm after ... updating a progress bar without the notification moving. That looks really nice. I'll have to dissect that `Start-Process.ps1` script to see what's happening.

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
